core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java

```diff
@@ -368,19 +373,34 @@ public void pruneWorkflowsAndTasks() {
     }
 
     /**
-     * Removes the workflow from the data store.
+     * Removes the workflow from the data store. Sets the removeFromIndex as true i.e default its
+     * trying to remove from index.
      *
      * @param workflowId the id of the workflow to be removed
      * @param archiveWorkflow if true, the workflow will be archived in the {@link IndexDAO} after
      *     removal from {@link ExecutionDAO}
      */
     public void removeWorkflow(String workflowId, boolean archiveWorkflow) {
+        removeWorkflow(workflowId, archiveWorkflow, false);
+    }
+
+    /**
+     * Removes the workflow from the data store.
+     *
+     * @param workflowId the id of the workflow to be removed
+     * @param archiveWorkflow if true, the workflow will be archived in the {@link IndexDAO}. Also
+     *     it will not be removed from datastore {@link ExecutionDAO}
+     */
+    public void removeWorkflow(
+            String workflowId, boolean archiveWorkflow, boolean isPollProcessing) {
```

Comment:
Can parameter `isPollProcessing` be renamed to `isIndexed` or something else ?
Poll processing name making it tied to fusion, for conductor it should not matter, is it by poll or push